src/containers/LoginPage.jsx

```diff
@@ -1,20 +1,44 @@
-// external libs
-import React, { Component } from 'react';
+import React, { Component, PropTypes } from 'react';
+import { connect } from 'react-redux';
+import { replace } from 'react-router-redux';
 
-// my libs (utils, actions)
+import { userLogin } from '../actions';
+import { getLoggingIn } from '../selectors/login';
 
-// external components
+import Login from '../components/Login.jsx';
 
-// my components
+@connect(mapStateToProps, { replace, userLogin })
+export default class LoginPage extends Component {
+    static propTypes = {
+        loggingIn: PropTypes.bool.isRequired,
+        replace: PropTypes.func.isRequired,
+        userLogin: PropTypes.func.isRequired
+    };
 
-// styles
+    componentWillReceiveProps(nextProps) {
+        if (nextProps.loggingIn) {
+            return;
```

Comment:
Strange moment. Should be OK after the next commit. 'if' required to skip route.replace in case if user still logging in.